### PR TITLE
schema_lk inversion

### PR DESCRIPTION
**Pull request by markhannum**
I've added instrumentation in hopes of being able to replicate the comdb2del bug brought to our attention by the DBA group.  This isn't ready to merge- I'm submitting the PR to see what RR tests fail.

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>2/374 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_timepart<br>maxtable</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/4a01db4/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/4a01db4/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2529'>Resubmit</a></p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>0/375 tests failed</b> &#x26A0;.</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/fcdea4d/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/fcdea4d/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2529'>Resubmit</a></p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>2/375 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>fingerprints<br>logfill</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/fcdea4d/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/fcdea4d/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2529'>Resubmit</a></p>

File: bdb/locks.c [hunk 1]
```diff
@@ -830,7 +830,35 @@ int bdb_release_row_lock(bdb_state_type *bdb_state, DB_LOCK *lk)
...
+}
+
+void bdb_locker_assert_nolocks(bdb_state_type *bdb_state, int lid)
+{
+    int nlocks = bdb_nlocks_for_locker(bdb_state, lid);
+    if (nlocks > 0) {
+        logmsg(LOGMSG_ERROR, "%s lockid %u holds %d locks\n", __func__, lid, nlocks);
+        bdb_lock_stats_me(bdb_state, stderr);
+    }
+    assert(nlocks == 0);
```
Comment (adizaimi) [inline bdb/locks.c:852]: just reminding you that the asserts will not kick in unless we build in Debug mode.

**Review by adizaimi — APPROVED** [review]
It looks good to me.

**Pull request merged**